### ISSUE TITLE
Create a Workflow suitable Docker container

### ISSUE THREAD

**Issue by johnbradley** [hdr-bgnn/drexel_metadata#7]
Create a docker container suitable for inclusion in a workflow for this repo that can be run on a single image that creates an output metadata file. Work has already been done on this in PR #6. I'm going to use this issue to document the changes related to creating the container so we can determine the minimal changes required to achieve this goal.

**Comment by johnbradley** [comment]
# Issue: `CUDA required`
When running `gen_metadata.py` on my laptop in a Docker container I received the following error:
```
Errored out (Torch not compiled with CUDA enabled)
```
I think this is due to the following code:
 https://github.com/hdr-bgnn/drexel_metadata/blob/a8720c3bfefcbd5caf7d948fad4ba90c2adc7f3c/gen_metadata.py#L43-L44
The `get_cfg()` function [defaults to using cuda](https://github.com/facebookresearch/detectron2/blob/2a1cec4c647dbe8a05f2d5b599d7d4c7cf9f04ad/detectron2/config/defaults.py#L32).

## Possible solution
The `mask_rcnn_R_50_FPN_3x.yaml` config file can be modified to use "cpu" instead of "cuda".
```
_BASE_: "Base-RCNN-FPN.yaml"
MODEL:
  WEIGHTS: "detectron2://ImageNetPretrained/MSRA/R-50.pkl"
  MASK_ON: True
  RESNETS:
    DEPTH: 50
  DEVICE: cpu
SOLVER:
  #STEPS: (210000, 250000)
  STEPS: (12300, 13500)
  MAX_ITER: 15000
```

**Comment by johnbradley** [comment]
## Issue: Metadata JSON printed and not saved
When run with a single file `gen_metadata.py` prints out the metadata to the terminal instead of saving it. Due to earlier printing commands the output cannot be simply redirected to a json file. Here is the code in question:

https://github.com/hdr-bgnn/drexel_metadata/blob/a8720c3bfefcbd5caf7d948fad4ba90c2adc7f3c/gen_metadata.py#L943-L947

`gen_metadata.py` is built around running on a batch of many files. I assume this behavior is for easy debugging of a problem with a single image.

### Possible solution
Introduce a new script named `gen_metadata_single.py` that re-uses the existing logic from `gen_metadata.py` but always processes a single input image file and saves the output to the specified output file.
```
import sys
import json
from gen_metadata import gen_metadata_safe

input_image_path=sys.argv[1]
output_metadata_path=sys.argv[2]
output = gen_metadata_safe(image_file)
with open(output_metadata, 'w') as outfile:
    json.dump(output, outfile)

```

**Comment by johnbradley** [comment]
## Problem: Provide a way to access mask data
The drexel_metadata logic creates a mask of the image but does not save this in the output files(as far as I can tell). 
There is some data from the mask stored in the metadata file at `fish/mask/*`. There is an `encoding` field, but I think that  is only for the longest `longest contour` in the image instead of the whole image.

**Comment by johnbradley** [comment]
In trying to most accurately reproduce the software environment I think I found an issue with the Pipfile.
When trying to install using pipenv I received the following error:
```
ERROR: Could not find a version that satisfies the requirement torch==1.10.1+cu113 (from versions: 1.11.0, 1.11.0+cu113, 1.12.0, 1.12.0+cu113, 1.12.1, 1.12.1+cu113)
```

---

The Pipfile specifies we should be using python 3.10 and install `torch` version `1.10.1`:
https://github.com/hdr-bgnn/drexel_metadata/blob/a8720c3bfefcbd5caf7d948fad4ba90c2adc7f3c/Pipfile#L34
https://github.com/hdr-bgnn/drexel_metadata/blob/a8720c3bfefcbd5caf7d948fad4ba90c2adc7f3c/Pipfile#L27

PyTorch version `1.10.1` only supports https://pypi.org/project/torch/1.10.1/

---
I think `python_version` was accidentally upgraded from `3.8.10` to `3.10` in this commit: https://github.com/hdr-bgnn/drexel_metadata/commit/c9667a6c31f075b05ae420ac291f2bf357843e34
For python `3.8.10` there is a `torch` version `1.10.1` that can be installed.

**Comment by johnbradley** [comment]
Fixed by PR #6
